### PR TITLE
test(shareable-resource): add unit testing for ui components (#2871)

### PR DESCRIPTION
**Pull request by Kroustille**
# Context
> Briefly describe the purpose and context of this PR. Include relevant screenshots or screen recordings for the product team.
>
> ℹ️ **Feature environment** — A dedicated preview environment (`https://dev-pr-{number}.hub.staging.filigran.io`) is automatically deployed when CI passes. Add the `skip-feature-env` label to opt out.

This PR adds frontend unit tests around the **shareable-resource UI components** (cards, lists, filters, and logical multi-select building blocks), improving confidence in rendering branches and callback wiring. It also refactors a few `react-hook-form` usages to use `useWatch` instead of `form.watch` for watched-field reads.

**Changes:**

- Added Vitest unit tests for Shareable Resource cards, public list rendering, and filter components (including logical multi-select internals).
- Added focused tests for shareable-resource card-design subcomponents (header/icon/image/description/footer/version).
- Replaced `form.watch(...)` with `useWatch(...)` in a few forms to read watched values safely.

Unit testing only, no needs to manually test this PR.

## Related issues

- Related to #2871

# Checklist

## Quality

- [x] I consider this work complete and ready for review
- [x] I tested all features and edge cases
- [x] I refactored code where necessary to improve quality
- [x] I made sure my code meets development guidelines
- [x] I performed a self-review of my code
- [x] I added the `skip-feature-env` label if a feature environment is not needed for this PR

## Testing

- [x] I wrote test cases for the relevant use cases (unit, integration or e2e tests)
- [ ] (Bug fixes only) I added tests that reproduce and verify the fix

### What tests have been made

- [x] Unit tests
- [ ] Integration tests
- [ ] E2E tests
- [ ] Local manual tests

# Additional information
> Any additional context, dependencies, or concerns reviewers should know about.

### CODE REVIEW

File: apps/frontend/src/components/ui/shareable-resource/ShareableResourceCard.test.tsx [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+vi.mock('@/hooks/use-scroll-position', () => ({
+  __esModule: true,
+  default: () => ({ save: saveMock }),
+}));
+
+vi.mock('@/hooks/use-is-feature-enabled', () => ({
+  useIsFeatureEnabled: () => useIsFeatureEnabledMock(),
+}));
+
+vi.mock('next/link', () => ({
```
Comment (hervyt) [inline apps/frontend/src/components/ui/shareable-resource/ShareableResourceCard.test.tsx]: **question**: Isn't it n other way to mock this ?

File: apps/frontend/src/components/ui/shareable-resource/ShareableResourceCard.test.tsx [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+  }) => (
+    <a
+      href={href}
+      onClick={onClick}>
+      {children}
+    </a>
+  ),
+}));
+
+vi.mock(
```
Comment (hervyt) [inline apps/frontend/src/components/ui/shareable-resource/ShareableResourceCard.test.tsx]: **question**: Do you really need to mock this subcomponent ? The less we mock, the more accurate we are :)

File: apps/frontend/src/components/ui/shareable-resource/ShareableResourceCard.test.tsx [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+  () => ({
+    ShareableResourceCardHeader: ({
+      shouldDisplayBothIcons,
+    }: {
+      shouldDisplayBothIcons: boolean;
+    }) => <div data-testid="card-header">{String(shouldDisplayBothIcons)}</div>,
+  })
+);
+
+vi.mock(
```
Comment (hervyt) [inline apps/frontend/src/components/ui/shareable-resource/ShareableResourceCard.test.tsx]: **question**: Do you really need to mock this subcomponent ? The less we mock, the more accurate we are :)

File: apps/frontend/src/components/ui/shareable-resource/ShareableResourceCard.test.tsx [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+  () => ({
+    ShareableResourceCardDescription: ({
+      description,
+    }: {
+      description?: string;
+    }) => <div data-testid="card-description">{description}</div>,
+  })
+);
+
+vi.mock('@/components/ui/BadgeOverflowCounter', () => ({
```
Comment (hervyt) [inline apps/frontend/src/components/ui/shareable-resource/ShareableResourceCard.test.tsx]: **question**: Do you really need to mock this subcomponent ? The less we mock, the more accurate we are :)

File: apps/frontend/src/components/ui/shareable-resource/ShareableResourceCard.test.tsx [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+  __esModule: true,
+  default: () => <div data-testid="badge-overflow">badges</div>,
+}));
+
+vi.mock(
```
Comment (hervyt) [inline apps/frontend/src/components/ui/shareable-resource/ShareableResourceCard.test.tsx]: **question**: Do you really need to mock this subcomponent ? The less we mock, the more accurate we are :)

File: apps/frontend/src/components/ui/shareable-resource/ShareableResourceCard.test.tsx [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+  '@/components/ui/shareable-resource/card-design/ShareableResourceCardFooterVersions',
+  () => ({
+    ShareableResourceCardFooterVersion: () => (
+      <div data-testid="footer-version" />
+    ),
+  })
+);
+
+vi.mock(
```
Comment (hervyt) [inline apps/frontend/src/components/ui/shareable-resource/ShareableResourceCard.test.tsx]: **question**: Do you really need to mock this subcomponent ? The less we mock, the more accurate we are :)

File: apps/frontend/src/components/ui/shareable-resource/PublicShareableResourceList.test.tsx [hunk 7]
```diff
@@ -0,0 +1,75 @@
+import testRender from '@/utils/test/test-render';
+import { IntegrationType } from '@graphql/generated';
+import { screen } from '@testing-library/react';
+import { describe, expect, it, vi } from 'vitest';
+import { PublicShareableResourceList } from './PublicShareableResourceList';
+
+vi.mock('@/components/ui/shareable-resource/ShareableResourceCard', () => ({
```
Comment (hervyt) [inline apps/frontend/src/components/ui/shareable-resource/PublicShareableResourceList.test.tsx]: **question**: Do you really need to mock this subcomponent ? The less we mock, the more accurate we are :)

File: apps/frontend/src/components/ui/shareable-resource/ProductVersionFilter.test.tsx [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+import { screen } from '@testing-library/react';
+import { describe, expect, it, vi } from 'vitest';
+import { ProductVersionFilter } from './ProductVersionFilter';
+
+const logicalMultiSelectPropsMock = vi.fn();
+const removeFilterMock = vi.fn();
+const setProductVersionsMock = vi.fn();
+const removeProductVersionsMock = vi.fn();
+
+vi.mock(
```
Comment (hervyt) [inline apps/frontend/src/components/ui/shareable-resource/ProductVersionFilter.test.tsx]: **question**: Do you really need to mock this subcomponent ? The less we mock, the more accurate we are :)

File: apps/frontend/src/components/ui/shareable-resource/logical-multi-select/SelectedValuesDisplay.test.tsx [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+            children: [{ value: 'c1', label: 'Child 1' }],
+          },
+          { parentValue: 'p2', parentLabel: 'Parent 2', children: [] },
+        ]}
+        optionLabel="Type"
+        placeholder="Choose values"
+      />
+    );
+
+    expect(screen.getAllByText('Type =').length).toBeGreaterThan(0);
```
Comment (hervyt) [inline apps/frontend/src/components/ui/shareable-resource/logical-multi-select/SelectedValuesDisplay.test.tsx]: **question**: Can't we have the correct numbers here ?

File: apps/frontend/src/components/ui/shareable-resource/logical-multi-select/SelectedValuesDisplay.test.tsx [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+vi.mock('@filigran/ui/clients', () => ({
+  TooltipProvider: ({ children }: { children: ReactNode }) => <>{children}</>,
+  Tooltip: ({ children }: { children: ReactNode }) => <>{children}</>,
+  TooltipTrigger: ({ children }: { children: ReactNode }) => <>{children}</>,
+  TooltipContent: ({ children }: { children: ReactNode }) => (
+    <div role="tooltip">{children}</div>
+  ),
+}));
+
+vi.mock('@filigran/ui/servers', () => ({
```
Comment (hervyt) [inline apps/frontend/src/components/ui/shareable-resource/logical-multi-select/SelectedValuesDisplay.test.tsx]: **question**: Do you really need to mock this subcomponent ? The less we mock, the more accurate we are :)

File: apps/frontend/src/components/ui/shareable-resource/logical-multi-select/SelectedValuesDisplay.test.tsx [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+import { screen } from '@testing-library/react';
+import type { ReactNode } from 'react';
+import { describe, expect, it, vi } from 'vitest';
+import { SelectedValuesDisplay } from './SelectedValuesDisplay';
+
+vi.mock('@filigran/icon', () => ({
+  CancelIcon: () => <svg data-testid="cancel-icon" />,
+}));
+
+vi.mock('@filigran/ui/clients', () => ({
```
Comment (hervyt) [inline apps/frontend/src/components/ui/shareable-resource/logical-multi-select/SelectedValuesDisplay.test.tsx]: **question**: Do you really need to mock this subcomponent ? The less we mock, the more accurate we are :)

File: apps/frontend/src/components/ui/shareable-resource/logical-multi-select/SelectedValuesDisplay.test.tsx [hunk 12]
```diff
@@ -0,0 +1,68 @@
+import testRender from '@/utils/test/test-render';
+import { screen } from '@testing-library/react';
+import type { ReactNode } from 'react';
+import { describe, expect, it, vi } from 'vitest';
+import { SelectedValuesDisplay } from './SelectedValuesDisplay';
+
+vi.mock('@filigran/icon', () => ({
```
Comment (hervyt) [inline apps/frontend/src/components/ui/shareable-resource/logical-multi-select/SelectedValuesDisplay.test.tsx]: **question**: Do you really need to mock this subcomponent ? The less we mock, the more accurate we are :)

File: apps/frontend/src/components/ui/shareable-resource/logical-multi-select/OptionsList.test.tsx [hunk 13]
```diff
@@ -0,0 +1,95 @@
+import testRender from '@/utils/test/test-render';
+import { screen } from '@testing-library/react';
+import type { ReactNode } from 'react';
+import { describe, expect, it, vi } from 'vitest';
+import { OptionsList } from './OptionsList';
+
+vi.mock('@filigran/ui', () => ({
```
Comment (hervyt) [inline apps/frontend/src/components/ui/shareable-resource/logical-multi-select/OptionsList.test.tsx]: **question**: Do you really need to mock this subcomponent ? The less we mock, the more accurate we are :)

File: apps/frontend/src/components/ui/shareable-resource/logical-multi-select/LogicalMultiSelectFormField.test.tsx [hunk 14]
```diff
@@ -0,0 +1,170 @@
...
+        <div>
+          <div data-testid="groups-count">{groupedSelections.length}</div>
+          <button onClick={onRemove}>remove-filter</button>
+        </div>
+      );
+    },
+  })
+);
+
+vi.mock(
```
Comment (hervyt) [inline apps/frontend/src/components/ui/shareable-resource/logical-multi-select/LogicalMultiSelectFormField.test.tsx]: **question**: Do you really need to mock this subcomponent ? The less we mock, the more accurate we are :)

File: apps/frontend/src/components/ui/shareable-resource/logical-multi-select/LogicalMultiSelectFormField.test.tsx [hunk 15]
```diff
@@ -0,0 +1,170 @@
...
+  }) => (
+    <input
+      aria-label="command-input"
+      onKeyDown={(event) => onKeyDown?.(event)}
+    />
+  ),
+  CommandList: ({ children }: { children: ReactNode }) => <div>{children}</div>,
+}));
+
+vi.mock('@filigran/ui/servers', () => ({
```
Comment (hervyt) [inline apps/frontend/src/components/ui/shareable-resource/logical-multi-select/LogicalMultiSelectFormField.test.tsx]: **question**: Do you really need to mock this subcomponent ? The less we mock, the more accurate we are :)

File: apps/frontend/src/components/ui/shareable-resource/card-design/ShareableResourceCardFooterAuthor.test.tsx [hunk 16]
```diff
@@ -0,0 +1,50 @@
...
+import { ShareableResourceCardFooterAuthor } from './ShareableResourceCardFooterAuthor';
+
+vi.mock('@/components/service/integrations/Integration.utils', () => ({
+  getIntegrationSubTypeMetadata: () => ({
+    label: 'Connector subtype',
+    color: '#123456',
+  }),
+}));
+
+vi.mock('@/utils/format/name', () => ({
```
Comment (hervyt) [inline apps/frontend/src/components/ui/shareable-resource/card-design/ShareableResourceCardFooterAuthor.test.tsx]: **question**: Still need this ? idk why you mock it

File: apps/frontend/src/components/ui/shareable-resource/card-design/ShareableResourceCardFooterVersions.test.tsx [hunk 17]
```diff
@@ -0,0 +1,53 @@
...
+
+  it('renders plain text when manager_supported is false', () => {
+    testRender(
+      <ShareableResourceCardFooterVersion
+        document={{ ...document, manager_supported: false } as never}
+        shareLinkUrl="https://share"
+      />
+    );
+
+    expect(screen.getByText('6.8')).toBeInTheDocument();
```
Comment (hervyt) [inline apps/frontend/src/components/ui/shareable-resource/card-design/ShareableResourceCardFooterVersions.test.tsx]: **nitpick**: We can extract it in a var and reuse this var in the document declaration to avoid this magic number (easier to read!)

File: apps/frontend/src/components/ui/shareable-resource/card-design/ShareableResourceCardFooterVersions.test.tsx [hunk 18]
```diff
@@ -0,0 +1,53 @@
...
+  it('renders version with share button when manager_supported is true', () => {
+    testRender(
+      <ShareableResourceCardFooterVersion
+        document={document as never}
+        shareLinkUrl="https://share"
+        extraContent={<span>extra</span>}
+      />
+    );
+
+    expect(screen.getByText('6.8')).toBeInTheDocument();
```
Comment (hervyt) [inline apps/frontend/src/components/ui/shareable-resource/card-design/ShareableResourceCardFooterVersions.test.tsx]: And reuse it here!

File: apps/frontend/src/components/ui/shareable-resource/card-design/ShareableResourceCardHeader.test.tsx [hunk 19]
```diff
@@ -0,0 +1,61 @@
+import testRender from '@/utils/test/test-render';
+import { screen } from '@testing-library/react';
+import { describe, expect, it, vi } from 'vitest';
+import { ShareableResourceCardHeader } from './ShareableResourceCardHeader';
+
+vi.mock('@/components/service/integrations/Integration.utils', () => ({
+  getIntegrationSubTypeMetadata: () => ({ label: 'Subtype label' }),
+}));
+
+vi.mock('next/image', () => ({
```
Comment (hervyt) [inline apps/frontend/src/components/ui/shareable-resource/card-design/ShareableResourceCardHeader.test.tsx]: No way to have a Next/Image in a component and test it easier ?

File: apps/frontend/src/components/ui/shareable-resource/card-design/ShareableResourceCardIcon.test.tsx [hunk 20]
```diff
@@ -0,0 +1,39 @@
...
+  it('for non-connector cards, renders the active icon only', () => {
+    const { container } = testRender(
+      <ShareableResourceCardIcon
+        shouldDisplayBothIcons={false}
+        document={
+          {
+            active: true,
+            manager_supported: true,
+            verified: true,
+          } as never
```
Comment (hervyt) [inline apps/frontend/src/components/ui/shareable-resource/card-design/ShareableResourceCardIcon.test.tsx]: Why don't you cast it with the correct type ?

File: apps/frontend/src/components/ui/shareable-resource/card-design/ShareableResourceCardIcon.test.tsx [hunk 21]
```diff
@@ -0,0 +1,39 @@
...
+  it('for connector-like cards, renders deployable and verified icons', () => {
+    const { container } = testRender(
+      <ShareableResourceCardIcon
+        shouldDisplayBothIcons
+        document={
+          {
+            active: true,
+            manager_supported: true,
+            verified: true,
+          } as never
```
Comment (hervyt) [inline apps/frontend/src/components/ui/shareable-resource/card-design/ShareableResourceCardIcon.test.tsx]: Correct type prefered here ?

File: apps/frontend/src/components/ui/shareable-resource/card-design/ShareableResourceCardHeader.test.tsx (hunk 19 above)
Comment (Kroustille) [reply apps/frontend/src/components/ui/shareable-resource/card-design/ShareableResourceCardHeader.test.tsx]: I've put a global mock for this one which renders a component

**Review by hervyt — APPROVED** [review]
